### PR TITLE
Allow Vector-likes in all operations

### PR DESCRIPTION
**Pull request by astronouth7303**
Make all operations work with vector-likes.

### CODE REVIEW

**Comment by pathunstrom** [comment]
I like where we're going with this.

Suggesting we put most of this under a class method `.from(value: Union[Sequence, Mapping[str, numbers]])` that just checks length then tries to instantiate and raises a Value error if it fails. Most of the logic is clear: Must be length 2, and look like a vector (in the case of maps), and then we can get a meaningful error everywhere, also provides a utility for people doing things programatically.

**Comment by astronouth7303** [comment]
Standard python is that conversion should be in the constructor, ie `Vector2({Vector-like})`, but otherwise yes.

**Comment by pathunstrom** [comment]
I only suggest a class method because the constructor is `Vector2(x, y)` and I want to keep that as a general rule. Class method as an alternative constructor is pretty well established pattern (see [`int.from_bytes`](https://docs.python.org/3/library/stdtypes.html#int.from_bytes)).

File: ppb_vector/vector2.py [hunk 1]
```diff
@@ -67,16 +84,18 @@ def __getitem__(self, item):
             raise TypeError
 
     def __repr__(self):
-        return "Vector2({}, {})".format(self.x, self.y)
+        return "{}({}, {})".format(type(self).__name__, self.x, self.y)
```
Comment (pathunstrom) [inline ppb_vector/vector2.py]: Since we're modifying this, lets switch to F-strings.

File: ppb_vector/vector2.py [hunk 2]
```diff
@@ -115,6 +134,7 @@ def reflect(self, surface_normal: 'Vector2') -> 'Vector2':
         """
         Calculate the reflection of the vector against a given surface normal
         """
+        surface_normal = _mkvector(surface_normal)
```
Comment (pathunstrom) [inline ppb_vector/vector2.py]: If we do this, we need to change the type hint.

On that front, we should probably build a union or type var for this.

**Comment by astronouth7303** [comment]
Addressed everything.

Not super happy with the name `.convert()`, but I couldn't come up with a better one.

**Comment by astronouth7303** [comment]
`from` is a keyword; can't be used as method names.

**Pull request merged**